src/utils/localization-helper.js

```diff
@@ -21,18 +19,22 @@ export const t = (key, locale, interpolation = null) => {
   let ongoing = '';
   let match = false;
   let lastIndex = 0;
+  let key = 0;
 
   /* eslint-disable no-cond-assign */
   while ((match = componentPlaceholdersReg.exec(translation))) {
+    key += 1;
     ongoing += translation.substring(lastIndex, match.index);
-    if (typeof interpolation[match[2]] === 'undefined') {
-      ongoing += this.config.missingPlaceholder;
-    } else if (typeof interpolation[match[2]] === 'object') {
+    const value = interpolation[match[2]];
```

Comment:
You can also add a local variable to save `typeof value`. And how about to replace the while loop with `translation .replace(componentPlaceholdersReg, (match) => { /* blabla */ })`?

Reply:
Done for the local `typeof value` const.

I didn't know that replace() also accepted a callback as a second parameter, named parameters greatly improve code readability, thank you!